File: 2018/PracticeSession/A-NumberGuessing.py
if __name__ == '__main__':
    T = int(input())
    for t in range(T):
        A, B = [int(x) + 1 for x in input().split()]
        N = int(input())
        answer = ""
        while answer != "CORRECT":
            guess = (A + B) // 2
            print(guess)
            answer = input()
            if answer == "TOO_SMALL":
                A = guess
            elif answer == "TOO_BIG":
                B = guess

File: 2018/PracticeSession/A-NumberGuessing.cpp
#include <bits/stdc++.h>

using namespace std;

int main() {
    int T, A, B, N, i, guess;
    string answer;
    cin >> T;
    for (int t = 1; t <= T; ++t) {
        cin >> A >> B;
        A++;
        B++;
        cin >> N;
        answer = "";
        while (answer != "CORRECT") {
            guess = (A + B) / 2;
            cout << guess << endl;
            cin >> answer;
            if (answer == "TOO_SMALL") {
                A = guess;
            } else if (answer == "TOO_BIG") {
                B = guess;
            }
        }
    }
    return 0;
}
